packages/forms/src/Components/BelongsToManyMultiSelect.php

```diff
@@ -71,7 +71,9 @@ public function relationship(string | Closure $relationshipName, string | Closur
             if ($component->hasOptionLabelFromRecordUsingCallback()) {
                 return $relationshipQuery
                     ->get()
-                    ->map(fn (Model $record) => $component->getOptionLabelFromRecord($record))
+                    ->mapWithKeys(fn (Model $record) => [
+                        $record->{$relatedKeyName} => $component->getOptionLabelFromRecord($record)
```

Comment:
Do we need `$relatedKeyName` here or can we use `->getKey()` like on the other components?

```suggestion
                        $record->getKey() => $component->getOptionLabelFromRecord($record)
```